File: Hello/app/src/main/java/org/texchtown/hello/MainActivity.java
package org.texchtown.hello;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onButton1Clicked(View V){
        Toast.makeText(this, "확인1 버튼을 눌렀어요.", Toast.LENGTH_LONG).show();
    }

    public void onButton2Clicked(View V) {
        Intent myIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://m.naver.com"));
        startActivity(myIntent);
    }


    public void onButton3Clicked(View V){
        Intent myIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("tel:010-1000-1000"));
        startActivity(myIntent);
    }

}

File: NaverBlog/app/src/main/java/org/texchtown/naverblog/MainActivity.kt
package org.texchtown.naverblog

import android.R
import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.view.Window
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import org.texchtown.naverblog.databinding.ActivityMainBinding
import org.texchtown.naverblog.databinding.CustomDialogBinding


class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var dialogBinding: CustomDialogBinding

    val TITLE = "title"
    val DETAIL = "detail"
    val SAVED_NUM = "current_page_num"
    val COUNT = "count"
    val AUTO = "auto_save"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.roundBtn5.setOnClickListener {
            //TODO dialog
            val intent = Intent(this, WriteActivity::class.java)
            if(detectAutoSave()){
                showCustomDialog()
            } else {
                startActivity(intent)
                finish()
            }
        }
    }

    private fun detectAutoSave() : Boolean{
        val sharedPreferences = getSharedPreferences(AUTO, MODE_PRIVATE)
        val title = sharedPreferences.getString(TITLE, "error")
        val detail = sharedPreferences.getString(DETAIL, "error")
        if(title == "error" && detail == "error"){
            return false
        }
        return true
    }


    //Function to display the custom dialog.
    private fun showCustomDialog() {
        val dialog = Dialog(this)
        //We have added a title in the custom layout. So let's disable the default title.
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        //The user will be able to cancel the dialog bu clicking anywhere outside the dialog.
        dialog.setCancelable(true)
        //Mention the name of the layout of your custom dialog.
        dialogBinding = CustomDialogBinding.inflate(layoutInflater)
        dialog.setContentView(dialogBinding.root)

        dialogBinding.btn2.setOnClickListener {
            val intent = Intent(this, WriteActivity::class.java)

            intent.putExtra(AUTO, true)
            startActivity(intent)
            dialog.dismiss()
            finish()
        }
        dialogBinding.btn1.setOnClickListener {
            dialog.dismiss()
        }
        dialog.show()
    }

}